### PR TITLE
Support `<<` and `>>`

### PR DESCRIPTION
**Pull request by jpoon**
Small PR that adds support for `>>` and `<<`.

### CODE REVIEW

File: src/mode/mode_command.ts [hunk 1]
```diff
@@ -17,23 +35,28 @@ export default class CommandMode extends Mode {
...
-            case 'l':
-                vscode.commands.executeCommand("cursorRight");
-                break;
+        var keyHandled = false;
+        for (var window = 1; window <= 2 && window <= this.keyHistory.length && keyHandled === false; window++) {
+            var keys : string  = _.takeRight(this.keyHistory, window).join('').toString();
+            
+            console.log(keys);
+            if (this.keyHandler[key] !== undefined) {
+                // Bug: 'two characters does not work'
```
Comment (jpoon) [inline src/mode/mode_command.ts]: There's a bug right now where for some reason, I can't build a key with more than one chars.

File: src/mode/mode_command.ts [hunk 2]
```diff
@@ -1,14 +1,27 @@
...
+            "k" : (position) => { vscode.commands.executeCommand("cursorUp"); },      
+            "l" : (position) => { vscode.commands.executeCommand("cursorRight"); },
+            ">>" : (position) => { vscode.commands.executeCommand("editor.action.indentLines"); },
+            "<<" : (position) => { vscode.commands.executeCommand("editor.action.outdentLines"); }, 
+        };
     }
 
     ShouldBeActivated(key : string, currentMode : ModeName) : boolean {
-        return (key === 'esc');
+        return (key === 'esc' || key == 'ctrl+[');
```
Comment (guillermooo) [inline src/mode/mode_command.ts]: When we have mappings, we can make `ctrl+[` one and expand it to `esc` somewhere upstream. So `esc` is the only "switch to normal mode" command. Just an idea for the future, though.

Comment (jpoon) [inline src/mode/mode_command.ts]: Absolutely! That is a much better way of doing that.

**Comment by jpoon** [comment]
@guillermooo: All the comments/commits from before has been lost.

I hadn't realized you were working out of this branch as well. I just `git push --force` it :(. In any case, I fixed the `key` vs `keys` typo.

File: src/mode/mode_normal.ts [hunk 3]
```diff
@@ -16,24 +29,24 @@ export default class CommandMode extends Mode {
...
-            case 'j':
-                vscode.commands.executeCommand("cursorDown");
-                break;
-            case 'k':
-                vscode.commands.executeCommand("cursorUp");
-                break;
-            case 'l':
-                vscode.commands.executeCommand("cursorRight");
+        var keyHandled = false;
+        for (var window = 0; window <= this.keyHistory.length; window++) {
```
Comment (guillermooo) [inline src/mode/mode_normal.ts]: Use `let` here because that's the intent?

File: src/mode/mode_normal.ts [hunk 4]
```diff
@@ -16,24 +29,24 @@ export default class CommandMode extends Mode {
...
+            var keysPressed = key + _.takeRight(this.keyHistory, window).join('');
```
Comment (guillermooo) [inline src/mode/mode_normal.ts]: var ==> let ?

File: src/mode/mode_normal.ts [hunk 5]
```diff
@@ -16,24 +29,24 @@ export default class CommandMode extends Mode {
...
+            
+            if (this.keyHandler[keysPressed] !== undefined) {
```
Comment (guillermooo) [inline src/mode/mode_normal.ts]: Would this be enough? `if (this.keyHandler[keysPressed]) ...`

File: src/mode/mode_normal.ts [hunk 6]
```diff
@@ -1,14 +1,27 @@
+import * as _ from 'lodash';
+import * as vscode from 'vscode';
 import {ModeName, Mode} from './mode';
 import {showCmdLine} from './../cmd_line/main';
-import * as vscode from 'vscode';
 
 export default class CommandMode extends Mode {
+    private keyHandler : { [key : string] : (position : vscode.Position) => void; } = {};
```
Comment (guillermooo) [inline src/mode/mode_normal.ts]: I'm unsure we'll ever need to pass the `position` explicitly. If we need it, we can retrieve it from the current document. Also, why pass just one position when there may be multiple carets?

Comment (jpoon) [inline src/mode/mode_normal.ts]: True, I'll get rid of that.

File: src/mode/mode_normal.ts (hunk 3 above)
Comment (jpoon) [inline src/mode/mode_normal.ts]: Yeah, habit. Should be adopting `let`.

**Pull request merged**